src/averageAges.js

```diff
@@ -36,8 +50,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let peopleArr = people.filter((person) => person.sex === 'f');
+  let sumOfAges = 0;
+
+  if (withChildren) {
+    peopleArr = peopleArr.filter((person) =>
+      people.find((item) => item.mother === person.name));
+  }
+
+  peopleArr.forEach((person) => {
```

Comment:
Replace the forEach loop with a reduce function to calculate the sum of ages. Example: `sumOfAges = peopleArr.reduce((sum, person) => sum + (person.died - person.born), 0);`

Reply:
also, you may move this to a separate function (where you are doing reduce and dividing value to the array length) and reuse it